application/src/main/java/org/togetherjava/tjbot/commands/utils/StringDistances.java

```diff
@@ -0,0 +1,146 @@
+            @NotNull CharSequence destination) {
+        // Given by the value in the last row and column
+        int[][] table = computeLevenshteinDistanceTable(source, destination);
+        int rows = table.length;
+        int columns = table[0].length;
+
+        return table[rows - 1][columns - 1];
+    }
+
+    /**
+     * Distance to receive a prefix of {@code destination} from {@code source} by editing that
+     * minimizes the distance.
+     *
+     * For example {@code prefixEditDistance("foa", "foobar")} is {@code 1}.
+     *
+     * @param source the source string to start with
+     * @param destination the destination string to receive a prefix of by editing the source
+     * @return the prefix edit distance
+     */
+    public static int prefixEditDistance(@NotNull CharSequence source,
+            @NotNull CharSequence destination) {
+        // Given by the smallest value in the last row
+        int[][] table = computeLevenshteinDistanceTable(source, destination);
+        int lastRowIndex = table.length - 1;
+
+        return Arrays.stream(table[lastRowIndex]).min().orElseThrow();
+    }
+
+    /**
+     * Computes the Levenshtein distance table for the given strings. See
+     * <a href="https://en.wikipedia.org/wiki/Levenshtein_distance">Levenshtein distance</a> for
+     * details.
+     *
+     * An example for {@code "abc"} to {@code "abcdefg"} would be:
+     * 
+     * <pre>
+     *   | 0 a b c d e f g
+     * -------------------
+     * 0 | 0 1 2 3 4 5 6 7
+     * a | 1 0 1 2 3 4 5 6
+     * b | 2 1 0 1 2 3 4 5
+     * c | 3 2 1 0 1 2 3 4
+     * </pre>
+     * 
+     * @param source the source string to start with
+     * @param destination the destination string to receive by editing the source
+     * @return the levenshtein distance table
+     */
+    private static int @NotNull [][] computeLevenshteinDistanceTable(@NotNull CharSequence source,
+            @NotNull CharSequence destination) {
+        int rows = source.length() + 1;
+        int columns = destination.length() + 1;
+        int[][] table = new int[rows][columns];
+
+        // Initialize first row and column for distances from the empty word to the target word
+        for (int y = 0; y < columns; y++) {
+            table[0][y] = y;
+        }
+        for (int x = 0; x < rows; x++) {
+            table[x][0] = x;
+        }
+
+        // Process row by row, selecting diagonal candidates
+        for (int x = 1; x < rows; x++) {
+            for (int y = 1; y < columns; y++) {
+                // Take minimum of all candidates
+                int upperCandidate = table[x - 1][y] + 1;
+                int leftCandidate = table[x][y - 1] + 1;
+                int diagonalCandidate = table[x - 1][y - 1];
+                if (source.charAt(x - 1) != destination.charAt(y - 1)) {
+                    diagonalCandidate++;
+                }
+
+                int bestCandidate = IntStream.of(upperCandidate, leftCandidate, diagonalCandidate)
```

Comment:
Inline this variable.

Reply:
i made this on purpose for readability, since i can then pick a meaningful name. this algorithm is not trivial for someone who doesnt know it and the variable names help understanding it and "visualizing it" on the table.

can you agree to that?